File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
// long long int ans=0;
// void sumRootToLeaf(TreeNode* root, string num){
//     if(!root) return;
//      if(!root->left && !root->right){
//             num+=to_string(root->val);
//            ans+=(stoi(num));
//            return;
//       }
//     num+=to_string(root->val);
//       sumRootToLeaf(root->left, num);
//       sumRootToLeaf(root->right, num);
// }
//     int sumNumbers(TreeNode* root) {
//         string str;
//         sumRootToLeaf(root,str);
//        return ans;
//     }

// int sumRootToLeaf(TreeNode* root, int sum){
//     if(!root) return 0;
//      if(!root->left && !root->right){
//              sum= sum*10+ root->val;
//            return sum;
//       }
//     sum= sum*10+ root->val;
//     return  sumRootToLeaf(root->left, sum) +  sumRootToLeaf(root->right, sum);
     
// }
//     int sumNumbers(TreeNode* root) {
        
//        return sumRootToLeaf(root,0);
//     }


int solve(TreeNode* root, int num){
if(!root) return 0;
if(!root->right && !root->left){
     num=num*10+root->val;
     return num;
}
    num=num*10+root->val;
    int l=solve(root->left,num);
    int r=solve(root->right,num);
    return l+r;
}

    int sumNumbers(TreeNode* root) {
        
       return solve(root,0);
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
function solve(root, num){
if(!root) return 0;
if(!root.right && !root.left){
     num=num*10+root.val;
     return num;
}
    num=num*10+root.val;
    const l=solve(root.left,num);
    const r=solve(root.right,num);
    return l+r;
}
    return solve(root,0);
};